ntoskrnl/mm/ARM3/sysldr.c

```diff
@@ -2964,7 +2964,52 @@ MmLoadSystemImage(IN PUNICODE_STRING FileName,
     PrefixName = *FileName;
 
     /* Check if we have a prefix */
-    if (NamePrefix) DPRINT1("Prefixed images are not yet supported!\n");
+    if (NamePrefix)
+    {
+        /* Check if "directory + prefix" is too long for the string */
+        if(PrefixName.MaximumLength + BaseDirectory.Length < PrefixName.MaximumLength)
+        {
+            Status = STATUS_INSUFFICIENT_RESOURCES;
+            goto Quickie;
```

Comment:
You aren't allocating chunk of memory here but the caller submitted an invalid argument. Use `STATUS_INVALID_PARAMETER`.

Reply:
I have changed the return values when integer overflow happens to `STATUS_INVALID_PARAMETER`.